core/src/main/scala/codesearch/core/search/Search.scala

```diff
@@ -14,28 +14,31 @@ import codesearch.core.regex.space.SpaceInsensitiveString
 import codesearch.core.search.Search.{CSearchResult, CodeSnippet, Package, PackageResult, snippetConfig}
 import codesearch.core.search.SnippetsGrouper.SnippetInfo
 import codesearch.core.util.Helper.readFileAsync
+import com.google.re2j._
 import fs2.{Pipe, Stream}
 import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
 import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
 
 import scala.sys.process.{Process, ProcessLogger}
-import com.google.re2j._
-
-import scala.util.{Success, Try}
 
 trait Search {
 
   protected def cindexDir: СindexDirectory
   protected def extensions: Extensions
   protected val logger: SelfAwareStructuredLogger[IO] = Slf4jLogger.unsafeCreate[IO]
 
-  def checkRegexpForValid(regexp: String): Try[Pattern] = {
-    Try(Pattern.compile(regexp))
+  def checkRegexpForValid(regexp: String): IO[Pattern] = {
+    IO { Pattern.compile(regexp) }
   }
 
-  def search(request: SearchRequest): IO[CSearchPage] = {
-    checkRegexpForValid(request.query) match {
-      case Success(_) => {
+  def search(request: SearchRequest): IO[Response] = {
+    checkRegexpForValid(request.query).attempt.flatMap {
+      case Left(error) => {
+        IO(
+          ErrorResponse(
+            error.getMessage.substring(0, 1).toUpperCase + error.getMessage.substring(1, error.getMessage.length)))
```

Comment:
Does not look very. Make for this a separate function with name like: `formatErrorMessage` or something like.

Reply:
@Kabowyad look at this also